pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -242,3 +243,65 @@ func stopCRCHyperkitProcess() error {
 	}
 	return nil
 }
+
+func getDaemonConfig() (*launchd.AgentConfig, error) {
+	logFilePath := filepath.Join(constants.CrcBaseDir, ".launchd-crcd.log")
+
+	env := map[string]string{"Version": version.GetCRCVersion()}
+	daemonConfig := launchd.AgentConfig{
+		Label:          constants.DaemonAgentLabel,
+		ExecutablePath: constants.CrcSymlinkPath,
+		StdOutFilePath: logFilePath,
+		StdErrFilePath: logFilePath,
+		Args:           []string{"daemon", "--log-level=debug"},
+		Env:            env,
+	}
+
+	return &daemonConfig, nil
+}
+
+func checkIfDaemonPlistFileExists() error {
+	daemonConfig, err := getDaemonConfig()
+	if err != nil {
+		return err
+	}
+	if err := launchd.CheckPlist(*daemonConfig); err != nil {
+		return err
+	}
+	if !launchd.AgentRunning(daemonConfig.Label) {
+		return fmt.Errorf("launchd agent '%s' is not running", daemonConfig.Label)
+	}
+	return nil
+}
+
+func fixDaemonPlistFileExists() error {
+	daemonConfig, err := getDaemonConfig()
+	if err != nil {
+		return err
+	}
+	return fixPlistFileExists(*daemonConfig)
+}
+
+func removeDaemonPlistFile() error {
+	if err := launchd.UnloadPlist(constants.DaemonAgentLabel); err != nil {
+		return err
+	}
+	return launchd.RemovePlist(constants.DaemonAgentLabel)
+}
+
+func fixPlistFileExists(agentConfig launchd.AgentConfig) error {
```

Comment:
Not sure it's worth its own helper since it is used only once, and `fixDaemonPlistFileExists` is short.